---
title: "DESeq2 and GSEA"
output: html_notebook
---


```{r}
 # installing DEseq2
if (!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")
  BiocManager::install("DESeq2")
library(DESeq2)

```


```{r}

setwd("~/Documents/MSc Genomics Data Science/Research Project (ML)/Project Data")

# reading in the raw read counts
 rawCounts <- read.csv("DSS_imp_transcriptomics_clusters.csv", row.names=1)
 
# eeplacing NA values with 0
rawCounts[is.na(rawCounts)] <- 0

# rounding the values in the dataframe
rawCounts <- as.data.frame(apply(rawCounts, 2, round))
 
head(rawCounts)
```

Reading in the sample mappings
```{r}
sampleData <- read.csv("sample_groups.csv", header = T)

sampleData
```

Creating the DEseq2DataSet object
```{r}
# measuring the effect of the Patient, controlling for Group differences - two factor design
dds <- DESeqDataSetFromMatrix(countData = rawCounts,
                             colData = sampleData,
                             design = ~ Group)

```


```{r}
mm <- model.matrix( ~ Group + Patient, colData(dds))

dds
```


```{r}
# filtering low count genes: ensuring at least 5 samples with a count of 10 or more
keep <- rowSums(counts(dds) >= 10) >= 5
dds <- dds[keep,]

```


Running DESeq algo
```{r}
dds <- DESeq(dds, full = mm)
```


Generating differential expression results
```{r}
res <- results(dds, contrast=list(c("Groupcluster2")))
summary(res)
```



```{r}
 # saving results to csv
 write.csv(as.data.frame(res), file="DE_Cluster1VCluster2_results.csv")
```


Most significantly expressed genes 
```{r}
res <- res[order(res$padj),]
head(res,10)

```

Loading DESeq results 
```{r}
DE.res <- read.csv("DE_Group1VGroup2_results_corrected2.csv", header = TRUE, row.names = 1)
DE.res
```


Plotting DESeq2 results in a volcano plot
```{r}
ggplot(DE.res, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(size = 1, aes(color = ifelse(padj < 0.05, ifelse(log2FoldChange > 0, "red", "blue"), "black"))) +
  labs(x = "log2 Fold Change", y = "-log10(Adjusted p-value)", title = "Diff Gene Expression of Drug-Resistant Group") +
  scale_color_manual(values = c("black", "blue", "red")) +
  theme_minimal() +
  theme(legend.position = "none")
```


GSEA Enrichment analysis 
```{r}
# loading packages
install.packages("data.table")
library(data.table)

BiocManager::install("fgsea")
library(fgsea) 
BiocManager::install("qusage")
library(qusage)
```


```{r}
# ranking DE results

# ordering DE results object by log2foldchange from high to low 

DE.res.ranked <- DE.res[order(DE.res$log2FoldChange, decreasing = T), ]

# creating a named vector of the ranks (with corresponding gene

DE.ranks <- setNames(DE.res.ranked$log2FoldChange, row.names(DE.res.ranked))
```


```{r}
# loading the Gene Set collection (C5 from GO, getting BP)

gmt.file <- read.gmt("c5.go.bp.v7.4.symbols.gmt") 

```


```{r}
# running fgsea and getting top 10 mostly differentially regulated BP gene sets

fgseaRes <- fgsea(gmt.file, DE.ranks, minSize=15, maxSize=500) 
```


```{r}
# getting top 10 enriched pathways 

top10_pathways <- head(fgseaRes[order(padj)],10)

top10_pathways
```


```{r}
# saving results to csv

top10 <- top10_pathways[, c("pathway", "pval", "padj", "log2err", "ES", "NES", "size")]

write.csv(top10, "top10GSEA_pathways.csv")
```

